Feature/readme (#29)
mercadopago/sdk-go: README.md

# Mercado Pago SDK for Go

[![Go Reference](https://pkg.go.dev/badge/github.com/mercadopago/sdk-go.svg)](https://pkg.go.dev/github.com/mercadopago/sdk-go)
[![License](https://img.shields.io/github/license/mercadopago/sdk-go)](https://github.com/mercadopago/sdk-go)

A comprehensive Go client library for integrating with the Mercado Pago API.

## 💡 Requirements

The SDK requires Go 1.21 or higher.

## 📲 Installation

First time using Mercado Pago? Create your [Mercado Pago account](https://www.mercadopago.com), if you don't have one already.

> 1. Install the Mercado Pago SDK for Go:

Reviewer: acho que podemos remover este 1. dado que só temos um passo agora 😅 
```suggestion
Install the Mercado Pago SDK for Go:
```

```go
	accessToken := "{{ACCESS_TOKEN}}"

	cfg, err := config.New(accessToken)
```

### Making API Requests

To make requests to the Mercado Pago APIs, you can use the packages provided by the SDK. For example, to list payment methods, you can use the `paymentmethod` package:

```go
	req := payment.Request{
		TransactionAmount: 105.1,
		PaymentMethodID:   "visa",
		Payer: &payment.PayerRequest{
			Email: "{{EMAIL}}",
		},
		Token:        "{{CARD_TOKEN}}",
		Installments: 1,
	}

	client := payment.NewClient(cfg)
        pay, err := client.Create(context.Background(), req)
```

### Exception throwing handling

Every package methods returns two variables: response (type of the package) and error (type of the std lib), which will contain any error thrown. It is important to handle these errors in the best possible way.
```go
	paymentMethods, err := client.List(context.Background())
	if err != nil {
		// appropriate treatment
	}
```

For more details on the available methods and request parameters, please refer to the [Go Reference](https://pkg.go.dev/github.com/mercadopago/sdk-go) documentation.

## 📚 Documentation

See our documentation for more details.

> - Mercado Pago API Reference: [English](https://www.mercadopago.com/developers/en/guides) | [Portuguese](https://www.mercadopago.com/developers/pt/guides) | [Spanish](https://www.mercadopago.com/developers/es/guides)

Reviewer: a referencia de API seria outra coisa, acho que aqui ficaria melhor algo como Mercado Pago Developers Guide
```suggestion
- Mercado Pago Developers Guide: [English](https://www.mercadopago.com/developers/en/guides) | [Portuguese](https://www.mercadopago.com/developers/pt/guides) | [Spanish](https://www.mercadopago.com/developers/es/guides)
```

# Mercado Pago SDK for Go

[![Go Reference](https://pkg.go.dev/badge/github.com/mercadopago/dx-go.svg)](https://pkg.go.dev/github.com/mercadopago/dx-go)
[![License](https://img.shields.io/apm/l/vim-mode)](https://github.com/mercadopago/sdk-go)

A comprehensive Go client library for integrating with the Mercado Pago API.

## 💡 Requirements

> The SDK requires Go 1.21 or higher.

Reviewer: bom, pelo uso de generics teria que ser pelo menos 1.18
podemos dar uma olhada se dá para ficar na 1.18, vou separar um tempo para baixar uma versao antiga anterior e tantar compilar

> The SDK requires Go 1.21 or higher.

Reviewer: podemos alterar para 1.x.x agora e quando lançarmos a v1 compilamos com a 1.18 para testar e colocamos o valor aqui, o que acham?

> The SDK requires Go 1.21 or higher.

Reviewer 2: por mim tudo bem

> The SDK requires Go 1.21 or higher.

Author: mantido então como `1.x.x`

```go
import "github.com/mercadopago/sdk-go"
```

That's it! The Mercado Pago SDK for Go has been successfully installed.

## 🌟 Getting Started

Simple usage looks like:

```go
package main

import (
	"context"
	"fmt"

	"github.com/mercadopago/sdk-go/pkg/config"
	"github.com/mercadopago/sdk-go/pkg/payment"
)

func main() {
	accessToken := "{{ACCESS_TOKEN}}"

	cfg, err := config.New(accessToken)
	if err != nil {
		fmt.Println(err)
		return
	}

	req := payment.Request{
		TransactionAmount: 105.1,
		PaymentMethodID:   "visa",
		Payer: &payment.PayerRequest{
			Email: "{{EMAIL}}",
		},
		Token:        "{{CARD_TOKEN}}",
		Installments: 1,
	}

	client := payment.NewClient(cfg)
	pay, err := client.Create(context.Background(), req)
	if err != nil {
		fmt.Println(err)
		return
	}

	fmt.Println(pay)
}
```

### SDK Configuration

Before making API requests, you need to initialize the SDK with your access token:

```go
	accessToken := "{{ACCESS_TOKEN}}"

	cfg, err := config.New(accessToken)
```

### Making API Requests

To make requests to the Mercado Pago API, you can use the methods provided by the SDK. For example, to create a payment, you can use the `Payment.Create` method:

> ```go
> 	req := payment.Request{
> 		TransactionAmount: 105.1,
> 		PaymentMethodID:   "visa",
> 		Payer: &payment.PayerRequest{
> 			Email: "{{EMAIL}}",
> 		},
> 		Token:        "{{CARD_TOKEN}}",
> 		Installments: 1,
> 	}
> 
> 	client := payment.NewClient(cfg)
> ```

Author: aqui a ideia seria ficar apenas a parte da chamada do serviço, não ter a parte de criação de client e configuração do AT. Nesse caso seria mudar somente por 

```suggestion
```go
	client := paymentmethod.NewClient(cfg)
	paymentMethods, err := client.List(context.Background())
```
```

certo?

> ```

Reviewer: certo!

### Exception throwing handling

> In both cases, client configuration and payment creation, the variable `err` is available, which will contain any error thrown, it is important to handle these errors in the best possible way

Reviewer 2: ```suggestion
Every package methods returns two variables: response (type of the package) and error (type of the std lib), which will contain any error thrown. It is important to handle these errors in the best possible way.```

```go

> 	if err != nil {
> 		fmt.Println(err)
> 		return
> 	}

Reviewer: ```suggestion
	paymentMethods, err := client.List(context.Background())
	if err != nil {
		// appropriate treatment
	}
```

```

For more details on the available methods and request parameters, please refer to the [Go Reference](https://pkg.go.dev/github.com/mercadopago/sdk-go) documentation.

## 📚 Documentation

See our documentation for more details.

> - Mercado Pago API Reference: [English](https://www.mercadopago.com/developers/en/guides)
> - Mercado Pago API Reference: [Spanish](https://www.mercadopago.com/developers/es/guides)

Reviewer 2: ```suggestion
- Mercado Pago API Reference: [English](https://www.mercadopago.com/developers/en/guides) | [Portuguese](https://www.mercadopago.com/developers/pt/guides) | [Spanish](https://www.mercadopago.com/developers/es/guides)
```

> - Mercado Pago API Reference: [English](https://www.mercadopago.com/developers/en/guides)
> - Mercado Pago API Reference: [Spanish](https://www.mercadopago.com/developers/es/guides)

Reviewer: ```suggestion
- Mercado Pago Developers Guide: [English](https://www.mercadopago.com/developers/en/guides)
- Mercado Pago Developers Guide: [Spanish](https://www.mercadopago.com/developers/es/guides)
- Mercado Pago Developers Guide: [Portuguese](https://www.mercadopago.com/developers/pt/guides)
```

## 🤝 Contributing

> All contributions are welcome, ranging from people wanting to triage issues, others wanting to write documentation, to people wanting to contribute code.

Reviewer 2: ```suggestion
All contributions are welcome, ranging from people wanting to triage issues, others wanting to write documentation, to people wanting to contribute with code.
```

Please read and follow our [contribution guidelines](CONTRIBUTING.md). Contributions not following these guidelines will be disregarded. The guidelines are in place to make all of our lives easier and make contribution a consistent process for everyone.

> ### Patches to Version 1.x.x

Reviewer: esta secao nao é necessária aqui

Author: é para remover a parte que fala sobre o `CONTRIBUTING.md`?

Reviewer: não, a parte "### Patches to Version 1.x.x"
a gente só vai ter a v1 por enquanto, não é uma nova versão, é um SDK novo

Since the release of version 2.0.0, version 1 is deprecated and will not be receiving new features, only bug fixes. If you need to submit PRs for that version, please do so by using `develop-v1` as your base branch.

## ❤️ Support

> If you require technical support, please contact our support team at our developers site: [English](https://www.mercadopago.com/developers/en/support/center/contact) / [Spanish](https://www.mercadopago.com/developers/es/support/center/contact)

Reviewer: falta o link em português

> ## 🏻 License

Reviewer: ainda temos que definir a licença

Author: licensa removida
